### ISSUE TITLE
[java] UseLocaleWithCaseConversions - false positive with a non-String class

### ISSUE THREAD

**Issue by winhkey** [pmd/pmd#4277]
**Affects PMD Version:** 6.53.0

**Rule:** [UseLocaleWithCaseConversions](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#uselocalewithcaseconversions)

**Description:** The class `Ulid` has a `toLowerCase()` method, but no overloads take a locale. The class `Ulid` is not a string.

**Code Sample demonstrating the issue:**

```java
public class Foo {
    public static String generateUlid(boolean uppercase) {
        Ulid ulid = Ulid.fast();
        return uppercase ? ulid.toString() : ulid.toLowerCase(); // here problem
    }
}
```

**Expected outcome:**

PMD reports a violation "UseLocaleWithCaseConversions: When doing a String.toLowerCase()/toUpperCase() call, use a Locale" at line 4, but that's wrong. That's a false positive.

**Running PMD through:** CLI

Additional info:

* Ulid comes from https://github.com/f4b6a3/ulid-creator
* Maven coordinates:
  ```xml
        <dependency>
            <groupId>com.github.f4b6a3</groupId>
            <artifactId>ulid-creator</artifactId>
            <version>5.1.0</version>
        </dependency>
  ```
* The source for the class `Ulid`: https://github.com/f4b6a3/ulid-creator/blob/dc0082bad6bc7d43f9a416971886f146ad26a4e2/src/main/java/com/github/f4b6a3/ulid/Ulid.java#L437

**Comment by jsotuyod** [comment]
This is no longer reproducible under PMD 7.0.0
